tidy3d/components/data/monitor_data.py

```diff
@@ -536,25 +538,30 @@ def mode_area(self) -> FreqModeDataArray:
         return FreqModeDataArray(area)
 
     def dot(
-        self, field_data: Union[FieldData, ModeSolverData], conjugate: bool = True
+        self, field_data: Union[FieldData, ModeData], conjugate: bool = True
```

Comment:
for the meantime should this accept either FieldData, ModeSolverData, OR ModeData?